### PR TITLE
[Keyboard] add itstleo9

### PR DESCRIPTION
**Pull request by ItsTLeo**
add itstleo9 to QMK Firmware source code

I made an orthor 3x3 firmware for my macropad, can be using with RP2040 and ATmega32u4. Help the one who dont know how to make a firmware just handwire, solder and flash my firmware

## Description

I made the handwired firmware that have QMK Configurator, Via and Vial keymap to make the people who don't know how to made a firmware just handwire the switches on correct pin, and then install the firmware to have a nice macropad. Also im using firmware with RP2040 and ATmega 32u4, two of the most popular and easy use Promicro.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/itstleo/itstleo9/itstleo9_32u4/readme.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# itstleo/handwired/it9_32u4
+
+This is my first handwire macropad project call itstleo9 or it9, using orthor 3x3 layout powered with 2 MCU are promicro ATmega32u4 and RasberryPi RP2040 for simple flash using QMK Toolbox and Drag&Drop file. Supported for Via and Vial keymap.
```
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_32u4/readme.md:3]:
```suggestion
Handwired 3x3 macropad built with Pro Micro compatible microcontrollers
```

File: keyboards/itstleo/itstleo9/itstleo9_32u4/readme.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+* Keyboard Maintainer: [Its TLeo](https://github.com/Its TLeo)
+* Hardware Supported: Promicro ATmega32u4
+* Hardware Availability: [Promicro](https://aliexpress.com/item/32888212119.html?spm=a2g0o.productlist.main.3.6e5ef70cdBVEF7&algo_pvid=fe4ec051-4a64-4e5d-810f-9984bfe852eb&aem_p4p_detail=2023070605140211214266150364960004273054&algo_exp_id=fe4ec051-4a64-4e5d-810f-9984bfe852eb-1&pdp_npi=3%40dis%21VND%21110929%2189788.0%21%21%214.67%21%21%402102160416886456429678477d07bc%2112000029855280287%21sea%21VN%210&curPageLogUid=FnJUdcJPhdXf&search_p4p_id=2023070605140211214266150364960004273054_2)
```
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_32u4/readme.md:7]:
```suggestion
* Hardware Availability: [Promicro](https://aliexpress.com/item/32888212119.html)
```

**Comment by dunk2k** [comment]
Could folder structure be changed to the following:
```
keyboards
└─ itstleo
   └─ itsfleo9
      ├─ keymaps
      │  ├─ default 
      │  │  └─ keymap.c
      │  └─ via
      │     └─ keymap.c
      ├─ rp2040
      │  ├─ info.json
      │  ├─ readme.md
      │  └─ rules.mk
      ├─ promicro
      │  ├─ info.json
      │  ├─ readme.md
      │  └─ rules.mk
      └─ info.json
```
respective `info.json` in rp2040 and promicro directories contain definitions pertinent to respective controller. Shared definitions go into `itstleo/itsfleo9/info.json`.

File: keyboards/itstleo/itstleo9/itstleo9_rp2040/info.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+    "manufacturer": "Its TLeo",
+    "keyboard_name": "itstleo/handwired/ItsTleo9",
```
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_rp2040/info.json:3]:
```suggestion
    "keyboard_name": "ItsTleo9",
```

File: keyboards/itstleo/itstleo9/itstleo9_rp2040/info.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        "cols": ["GP10", "GP11", "GP12"],
+        "rows": ["GP21", "GP20", "GP19"]
+    },
+    "processor": "RP2040",
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0001",
+        "vid": "0x4954"
+    },
```
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_rp2040/info.json:25]:
```suggestion
    },
    "community_layouts": ["ortho_3x3"],
```

File: keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/via/config.h
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/via/config.h:1]: This file should be removed.

File: keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/via/itstleo9_rp2040.json
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/via/itstleo9_rp2040.json:1]: This file should be removed.

File: keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/via/rules.mk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# This file intentionally left blank
```
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/via/rules.mk:1]: As above.

File: keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/vial/config.h
Comment (waffle87) [inline keyboards/itstleo/itstleo9/itstleo9_rp2040/keymaps/vial/config.h:1]: This directory needs to be removed.

**Review by waffle87 — CHANGES_REQUESTED** [review]
This keyboard's file structure should be consolidated into one:
```shell
itstleo9
├── atmega32u4
│   └── info.json # avr specific configuration
├── info.json # global configuration that applies to all hardware (eg. layout)
├── keymaps
└── rp2040
    └── info.json # rp2040 specific configuration
```

File: keyboards/itstleo/itstleo9/itstleo9_32u4/config.h
[the remark this answers is not in the document]
Comment (ItsTLeo) [reply keyboards/itstleo/itstleo9/itstleo9_32u4/config.h:1]: i have change all, hope you check it again. Thank you so much for fixing my source code

**Comment by dunk2k** [comment]
Linting check failing as directory names must be all lower case

File: keyboards/itstleo/itstleo9/keymaps/via/itstleo9.json
Comment (dunk2k) [inline keyboards/itstleo/itstleo9/keymaps/via/itstleo9.json:1]: Could you remove this file from this PR and instead include it in a PR to VIA's [keyboards](https://github.com/the-via/keyboards) GitHub repo.

File: keyboards/itstleo/itstleo9/Promicro/readme.md
Comment (waffle87) [inline keyboards/itstleo/itstleo9/Promicro/readme.md:1]: Could you move this file up to the keyboard level (`keyboards/itstleo/itstleo9`)

File: keyboards/itstleo/itstleo9/RP2040/readme.md
Comment (waffle87) [inline keyboards/itstleo/itstleo9/RP2040/readme.md:1]: Then remove this file.

File: keyboards/itstleo/itstleo9/info.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0001",
+        "vid": "0x4954"
+    },
```
Comment (waffle87) [inline keyboards/itstleo/itstleo9/info.json:19]:
```suggestion
    },
    "community_layouts": ["ortho_3x3"],
```

File: keyboards/itstleo/itstleo9/keymaps/via/rules.mk [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# This file intentionally left blank
+VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (waffle87) [inline keyboards/itstleo/itstleo9/keymaps/via/rules.mk:3]:
```suggestion
VIA_ENABLE = yes
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/itstleo/itstleo9/keymaps/via/rules.mk (hunk 7 above)
Comment (ItsTLeo) [reply keyboards/itstleo/itstleo9/keymaps/via/rules.mk]: all changed, thank you

**Comment by ItsTLeo** [comment]
finished all the change that need to fix on this itstleo9 source code

**Comment by ItsTLeo** [comment]
> Handwired keyboards should be placed under `keyboards/handwired`.

i will change it as fast as i can, thankyou for remind me

**Comment by ItsTLeo** [comment]
changed the sourcecode path

File: keyboards/handwired/itstleo9/readme.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
+# itstleo/handwired/it9_rp2040
```
Comment (waffle87) [inline keyboards/handwired/itstleo9/readme.md:1]:
```suggestion
# itstleo9
```

File: keyboards/handwired/itstleo9/readme.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+Handwired 3x3 macropad built with Pro Micro compatible microcontrollers
+
+* Keyboard Maintainer: [ItsTLeo](https://github.com/ItsTLeo)
+* Hardware Supported: RP2040, Promicro ATmega32u4
+* Hardware Availability: [RP2040](https://aliexpress.com/item/1005005616524430.html?gatewayAdapt=glo2vnm),
+                         [Promicro](https://aliexpress.com/item/32888212119.html)
```
Comment (waffle87) [inline keyboards/handwired/itstleo9/readme.md:8]: Also, neither of these links seem to work..
```suggestion
* Hardware Supported: RP2040/ATmega32U4 Pro Micro
* Hardware Availability: [RP2040](https://aliexpress.com/item/1005005616524430.html), [ATmega32U4](https://aliexpress.com/item/32888212119.html)
```

File: keyboards/handwired/itstleo9/readme.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make itstleo/itstleo9:default
+
+Flashing example for this keyboard:
+
+    make itstleo/itstleo9:default:flash
```
Comment (waffle87) [inline keyboards/handwired/itstleo9/readme.md:16]:
```suggestion
    make handwired/itstleo9/promicro:default
    make handwired/itstleo9/rp2040:default
    
Flashing example for this keyboard:

    make handwired/itstleo9/promicro:default:flash
    make handwired/itstleo9/rp2040:default:flash
```

File: keyboards/handwired/itstleo9/readme.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+* **Physical reset button**: **Hold boot button** and press **rst button** 2 time, when it's appear the folder of rp2040 you entered the bootloader mode
+* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
+
+## Bootloader Promicro
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```
Comment (waffle87) [inline keyboards/handwired/itstleo9/readme.md:48]:
```suggestion
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (top left key) and plug in the keyboard
* **Physical reset button**: Double-press reset button
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by ItsTLeo** [comment]
thankyou all guys

**Review by waffle87 — APPROVED** [review]

**Review by dunk2k — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**